### PR TITLE
[WiP] Add PKCS#11 support with SoftHSM integration

### PR DESCRIPTION
**Pull request by FHatCSW**
- Add `Pkcs11Utilities` class with key management, session management, token handling, and random generation utilities.
- Introduce `PKCS11Token` model in Django for token metadata representation.
- Add initial setup for SoftHSM configuration in Dockerfile and volumes in `docker-compose.yml`.
- Include tests for `Pkcs11Utilities` and `Pkcs11PrivateKey` functionalities, covering initialization, caching, sessions, and edge cases.
- Update dependencies to include `python-pkcs11` and `asn1crypto`.

**Legal** 
- [ ] I certify that I have all necessary rights to publish this contribution under the MIT license. I agree to the Trustpoint CLA and have added my name to the AUTHORS.md file.

### CODE REVIEW

File: docker/trustpoint/Dockerfile [hunk 1]
```diff
@@ -23,8 +25,22 @@ RUN apt-get update && \
...
+        postgresql-client \
+        git \
+        build-essential \
+        cmake \
+        libssl-dev \
+        libseccomp-dev \
+        && rm -rf /var/lib/apt/lists/*
+
+# Clone and build pkcs11-proxy
+RUN git clone https://github.com/SUNET/pkcs11-proxy.git /tmp/pkcs11-proxy && \
```
[the remark this answers is not in the document]
Comment (FHatCSW) [reply docker/trustpoint/Dockerfile]: There are already some maintained alternatives (forks):
- https://github.com/bukka/pkcs11-proxy?utm_source=chatgpt.com

This may also be an option:
- https://pkg.go.dev/github.com/porech/pkcs11-web-proxy?utm_source=chatgpt.com
